src/wallet/walletdb.cpp

```diff
@@ -124,7 +124,25 @@ bool CWalletDB::WriteCryptedZKey(const libzcash::SproutPaymentAddress & addr,
     }
     return true;
 }
+bool CWalletDB::WriteCryptedSaplingZKey(const libzcash::SaplingPaymentAddress &addr,
+                          const libzcash::SaplingFullViewingKey &fvk,
+                          const std::vector<unsigned char>& vchCryptedSecret,
+                          const CKeyMetadata &keyMeta)
+{
+    const bool fEraseUnencryptedKey = true;
+    nWalletDBUpdated++;
 
+    if (!Write(std::make_pair(std::string("sapzkeymeta"), addr), keyMeta))
+        return false;
+
+    if (!Write(std::make_pair(std::string("csapzkey"), addr), std::make_pair(fvk, vchCryptedSecret), false))
```

Comment:
`if (!Write(std::make_pair(std::string("csapzkey"), ivk), std::make_pair(fvk, vchCryptedSecret), false))` (and get `ivk` from `fvk`).